src/Std/Time/Time/PlainTime.lean

```diff
@@ -0,0 +1,244 @@
+  time.hour.snd.toOffset.toMinutes +
+  time.minute.toOffset +
+  time.second.snd.toOffset.toMinutes
+
+/--
+Converts a `PlainTime` value to the total number of hours.
+-/
+def toHours (time : PlainTime) : Hour.Offset :=
+  let hour : Hour.Offset := time.minute.toOffset.ediv 60
+  time.hour.snd.toOffset + hour + time.second.snd.toOffset.toHours
+
+/--
+Creates a `PlainTime` value from a total number of nanoseconds.
+-/
+def ofNanoseconds (nanos : Nanosecond.Offset) : PlainTime :=
+  have totalSeconds := nanos.ediv 1000000000
+  have remainingNanos := Bounded.LE.byEmod nanos.val 1000000000 (by decide)
+  have hours := Bounded.LE.byEmod (totalSeconds.val / 3600) 24 (by decide)
+  have minutes := (Bounded.LE.byEmod totalSeconds.val 3600 (by decide)).ediv 60 (by decide)
+  have seconds := Bounded.LE.byEmod totalSeconds.val 60 (by decide)
+  let nanos := Bounded.LE.byEmod nanos.val 1000000000 (by decide)
+  ofValidHourMinuteSecondsNano hours minutes seconds nanos
+
+/--
+Creates a `PlainTime` value from a total number of seconds.
+-/
+@[inline]
+def ofSeconds (secs : Second.Offset) : PlainTime :=
+  have hours := Bounded.LE.byEmod (secs.val / 3600) 24 (by decide)
+  have minutes := (Bounded.LE.byEmod secs.val 3600 (by decide)).ediv 60 (by decide)
+  have seconds := Bounded.LE.byEmod secs.val 60 (by decide)
+  ofValidHourMinuteSecondsNano hours minutes seconds 0
+
+/--
+Adds seconds to a `PlainTime`.
+-/
+@[inline]
+def addSeconds (time : PlainTime) (secondsToAdd : Second.Offset) : PlainTime :=
+  let totalSeconds := time.toSeconds + secondsToAdd
+  ofSeconds totalSeconds
+
+/--
+Subtracts seconds from a `PlainTime`.
+-/
+@[inline]
+def subSeconds (time : PlainTime) (secondsToSub : Second.Offset) : PlainTime :=
+  addSeconds time (-secondsToSub)
+
+/--
+Adds minutes to a `PlainTime`.
+-/
+@[inline]
+def addMinutes (time : PlainTime) (minutesToAdd : Minute.Offset) : PlainTime :=
+  let total := time.toSeconds + minutesToAdd.toSeconds
+  ofSeconds total
+
+/--
+Subtracts minutes from a `PlainTime`.
+-/
+@[inline]
+def subMinutes (time : PlainTime) (minutesToSub : Minute.Offset) : PlainTime :=
+  addMinutes time (-minutesToSub)
+
+/--
+Adds hours to a `PlainTime`.
+-/
+def addHours (time : PlainTime) (hoursToAdd : Hour.Offset) : PlainTime :=
+  let total := time.toSeconds + hoursToAdd.toSeconds
+  ofSeconds total
+
+/--
+Subtracts hours from a `PlainTime`.
+-/
+@[inline]
+def subHours (time : PlainTime) (hoursToSub : Hour.Offset) : PlainTime :=
+  addHours time (-hoursToSub)
+
+/--
+Adds nanoseconds to a `PlainTime`.
+-/
+def addNanoseconds (time : PlainTime) (nanosToAdd : Nanosecond.Offset) : PlainTime :=
+  let total := time.toNanoseconds + nanosToAdd
+  ofNanoseconds total
+
+/--
+Subtracts nanoseconds from a `PlainTime`.
+-/
+def subNanoseconds (time : PlainTime) (nanosToSub : Nanosecond.Offset) : PlainTime :=
+  addNanoseconds time (-nanosToSub)
+
+instance : HAdd PlainTime Nanosecond.Offset PlainTime where
```

Comment:
I think there should be `HAdd PlainTime Duration PlainTime`, and similar for `PlainDateTime`. Unsure about `PlainDate`.